packages/driver/src/cy/net-stubbing/events/before-request.ts

```diff
@@ -121,6 +121,33 @@ export const onBeforeRequest: HandlerFn<CyHttpMessages.IncomingRequest> = (Cypre
 
   const userReq: CyHttpMessages.IncomingHttpRequest = {
     ...req,
+    get query () {
+      const url = new URL(req.url)
+      const urlSearchParams = new URLSearchParams(url.search)
+      const result = {}
+
+      for (let pair of urlSearchParams.entries()) {
+        result[pair[0]] = pair[1]
+      }
+
+      return result
```

Comment:
In the current implementation, I don't think I understand why it's necessary to do this, is it wrong to just return `urlSearchParams` instead of this `result` object?

Reply:
The returned type `URLSearchParams` isn't a plain object type. Because of that, we cannot do our everyday things like `req.query.someKey`.